### ISSUE TITLE
Make client, server and shared into modules

### ISSUE THREAD

**Issue by koraa** [inexor-game/code#170]
At the moment those are special; those are by listing the objects in the server/client – which is bad

**Comment by a-teammate** [comment]
what?

**Comment by koraa** [comment]
@a-teammate There is a list of object files for the client and the server. I'd like to get rid of *all* explicit source file lists.
Also shared/ is not a proper module (built as shared lib) at the moment but instead we compile the shared stuff twice: Once for the client, once for the server.

**Comment by aschaeffer** [comment]
Thx for clearification. I've changed the topic again.

**Comment by Croydon** [comment]
How do you want to get rid of the source file lists?
> Note: We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. 

https://cmake.org/cmake/help/v3.4/command/file.html

**Comment by Croydon** [comment]
Since we will likely never get an answer from @koraa concerning this issue I think we should close it as wontfix.

Getting rid of source lists doesn't seem to be possible in a sophisticated way (see my last comment).

**Comment by a-teammate** [comment]
I dont think so, the issue is still valid and wont get closed until a lot more refractorings (when we fully got rid of the initial 3 directories) are done.
the part u quoted is not an issue for us (it does work for me)
